src/platform_impl/macos/window.rs

```diff
@@ -452,29 +448,28 @@ impl WinitWindow {
         })
         .ok_or_else(|| os_error!(OsError::CreationError("Couldn't create `NSWindow`")))?;
 
-        match attrs.parent_window {
-            Some(RawWindowHandle::AppKit(handle)) => {
+        #[cfg(feature = "rwh_06")]
+        match attrs.parent_window.map(|parent| parent.as_raw()) {
+            Some(rwh_06::RawWindowHandle::AppKit(handle)) => {
                 // SAFETY: Caller ensures the pointer is valid or NULL
-                let parent: Id<NSWindow> = match unsafe { Id::retain(handle.ns_window.cast()) } {
-                    Some(window) => window,
-                    None => {
-                        // SAFETY: Caller ensures the pointer is valid or NULL
-                        let parent_view: Id<NSView> =
-                            match unsafe { Id::retain(handle.ns_view.cast()) } {
-                                Some(view) => view,
-                                None => {
-                                    return Err(os_error!(OsError::CreationError(
-                                        "raw window handle should be non-empty"
-                                    )))
-                                }
-                            };
-                        parent_view.window().ok_or_else(|| {
-                            os_error!(OsError::CreationError(
-                                "parent view should be installed in a window"
-                            ))
-                        })?
-                    }
+                let parent = {
+                    // SAFETY: Caller ensures the pointer is valid or NULL
+                    let parent_view: Id<NSView> =
+                        match unsafe { Id::retain(handle.ns_view.as_ptr().cast()) } {
+                            Some(view) => view,
+                            None => {
+                                return Err(os_error!(OsError::CreationError(
+                                    "raw window handle should be non-empty"
+                                )))
+                            }
+                        };
+                    parent_view.window().ok_or_else(|| {
+                        os_error!(OsError::CreationError(
+                            "parent view should be installed in a window"
+                        ))
+                    })?
                 };
```

Comment:
Nit: Could be simplified to something like this instead:
```suggestion
                // SAFETY: The safety contract of WindowHandle<'_> ensures the pointer is valid
                // Unwrap is fine, since the pointer comes from `NonNull`.
                let parent_view: Id<NSView> = unsafe { Id::retain(handle.ns_view.as_ptr().cast()) }.unwrap();
                let parent = parent_view.window().ok_or_else(|| {
                    os_error!(OsError::CreationError(
                        "parent view should be installed in a window"
                    ))
                })?;
```